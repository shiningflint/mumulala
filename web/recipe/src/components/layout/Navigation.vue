<template>
  <div>
    <h3>Navigation</h3>
    <ul class="links">
      <li v-for="link in links">
        <router-link :to="link.to">{{ link.label }}</router-link>
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  data () {
    return {
      links: [
        { to: { name: 'home', }, label: 'Home', },
        { to: { name: 'about', }, label: 'About', },
      ],
    }
  },
}
</script>

<style lang="scss" scoped>
.links {
  list-style: none;
  margin: 0;
  padding: 0;
}
</style>
