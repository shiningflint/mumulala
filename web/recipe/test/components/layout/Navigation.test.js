import { shallowMount, createLocalVue } from '@vue/test-utils'
import Navigation from 'components/layout/Navigation'

const localVue = createLocalVue()

describe('Navigation', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Navigation, {
      localVue
    })
  })

  it('Mounts without error', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
