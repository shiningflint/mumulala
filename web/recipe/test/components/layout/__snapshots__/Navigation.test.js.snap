// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Mounts without error 1`] = `
<div>
  <h3>
    Navigation
  </h3>
   
  <ul
    class="links"
  >
    <li>
      <router-link
        to="[object Object]"
      >
        Home
      </router-link>
    </li>
    <li>
      <router-link
        to="[object Object]"
      >
        About
      </router-link>
    </li>
  </ul>
</div>
`;
